### PR TITLE
[CI:DOCS] Add more information and examples on podman and pipes

### PR DESCRIPTION
**Pull request by rhatdan**
Improve the documentation to help users to know proper way to
use podman within a pipe.

Helps Prevent: https://github.com/containers/podman/issues/8916

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-run.1.md [hunk 1]
```diff
@@ -1011,8 +1011,14 @@ When set to **true**, Podman will allocate a pseudo-tty and attach to the standa
...
 
-**NOTE**: The **-t** option is incompatible with a redirection of the Podman client
-standard input.
+**NOTE**: The **-t** option is incompatible with a redirection of the Podman client standard input.
+
+```
+Use
+echo "asdf" | podman run --rm -i someimage /bin/cat
+Versus
+echo "asdf" | podman run --rm -it someimage /bin/cat
```
Comment (edsantiago) [inline docs/source/markdown/podman-run.1.md:1020]: Both sample commands have equal visual prominence, making it hard for my eye to figure out which is the correct one. I'm a fan of "show what to do, and hide the taboos". Could I suggest something like:

...incompatible with a redirection of the Podman client standard input. That is, when feeding input to podman, use `-i` only, not `-it`:
```
    $ echo asdf | podman run --rm -i someimage /bin/cat
```

File: docs/source/markdown/podman-run.1.md [hunk 2]
```diff
@@ -1528,6 +1534,13 @@ weight by **--blkio-weight-device** flag. Use the following command:
 $ podman run -it --blkio-weight-device "/dev/sda:200" ubuntu
 ```
 
+### Using a podman container from within a piped command
```
Comment (edsantiago) [inline docs/source/markdown/podman-run.1.md:1537]: I'm not entirely sure this title reflects the intention. I find it hard to parse. Maybe "Using a podman container *with input from a pipe*"? Or "Piping command output to a podman container"?

**Review by edsantiago — COMMENTED** [review]
LGTM but I'm picky; feel free to ignore my nits.

**Comment by mheon** [comment]
We should also mention it prevents redirection of standard output (combines
STDOUT and STDERR, can insert control characters, can hang pipes).

File: docs/source/markdown/podman-run.1.md [hunk 3]
```diff
@@ -1011,8 +1011,11 @@ When set to **true**, Podman will allocate a pseudo-tty and attach to the standa
 input of the container. This can be used, for example, to run a throwaway
 interactive shell. The default is **false**.
 
-**NOTE**: The **-t** option is incompatible with a redirection of the Podman client
-standard input.
+**NOTE**: The --tty flag prevents redirection of standard output (combines STDOUT and STDERR, can insert control characters, can hang pipes. It should only be used when run interactivley in a terminal. When feeding input to Podman, use -i only, not -it.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1014]: missing an end paren,  Maybe?
```suggestion
**NOTE**: The --tty flag prevents redirection of standard output.  It combines STDOUT and STDERR, it can insert control characters, and it can hang pipes. This option should only be used when run interactively in a terminal. When feeding input to Podman, use -i only, not -it.
```

**Comment by TomSweeneyRedHat** [comment]
LGTM
